api/utils.ts

```diff
@@ -0,0 +1,21 @@
+export function allDoNotContainDangerousCharacters(values: string[]) {
+  for (const value of values) {
+    if (!hasNoDangerousCharacters(value)) {
+      // if one value contains dangerous characters, return false
+      return false;
+    }
+  }
+  return true;
+}
+
+export function hasNoDangerousCharacters(value: string): boolean {
+  // the below only checks for the characters in the string
+  // return /^[^%<>^$//]+$/.test(value);
+  // substring approach below preferred
```

Comment:
i don't understand "substring approach below preferred"

Reply:
Will describe in response to comment below